frontend/catalyst/device/qjit_device.py

```diff
@@ -200,7 +200,7 @@ def extract_backend_info(device: qml.QubitDevice, capabilities: DeviceCapabiliti
             )
 
     for k, v in capabilities.options.items():
-        if hasattr(device, v):
+        if hasattr(device, v) and not k in device_kwargs:
             device_kwargs[k] = getattr(device, v)
```

Comment:
What scenario does this cover? Is it problematic to reassign the value?

Reply:
This is for a case where the shots was included in the capabilities.options and this would reassign the shots that was already set by get_device_shots to the unprocessed shots that causes the issue.